<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ListWidgetandQToolButton</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonTextUnderIcon</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QToolBox" name="toolBox">
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>200</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <widget class="QWidget" name="page">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>152</width>
          <height>391</height>
         </rect>
        </property>
        <attribute name="icon">
         <iconset>
          <normaloff>icons/410.bmp</normaloff>icons/410.bmp</iconset>
        </attribute>
        <attribute name="label">
         <string>列表项操作</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0">
          <widget class="QToolButton" name="tBtnListInit">
           <property name="text">
            <string>tBtnListInit</string>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
           <property name="autoRaise">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QToolButton" name="tBtnListClear">
           <property name="text">
            <string>tBtnListClear</string>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
           <property name="autoRaise">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QToolButton" name="tBtnListInsert">
           <property name="text">
            <string>tBtnListInsert</string>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
           <property name="autoRaise">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QToolButton" name="tBtnListAppend">
           <property name="text">
            <string>tBtnListAppend</string>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
           <property name="autoRaise">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QToolButton" name="tBtnListDelete">
           <property name="text">
            <string>tBtnListDelete</string>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
           <property name="autoRaise">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="widget">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>150</width>
          <height>408</height>
         </rect>
        </property>
        <attribute name="icon">
         <iconset>
          <normaloff>icons/408.bmp</normaloff>icons/408.bmp</iconset>
        </attribute>
        <attribute name="label">
         <string>列表排序</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="2" column="0">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="5" column="0">
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="chkAllowSort">
           <property name="text">
            <string>允许排序</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QToolButton" name="tBtnUP">
           <property name="text">
            <string>升序排序</string>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
           <property name="autoRaise">
            <bool>true</bool>
           </property>
           <property name="arrowType">
            <enum>Qt::UpArrow</enum>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QToolButton" name="tBtnDown">
           <property name="text">
            <string>降序排序</string>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
           <property name="autoRaise">
            <bool>true</bool>
           </property>
           <property name="arrowType">
            <enum>Qt::DownArrow</enum>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <spacer name="verticalSpacer_4">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="page_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>181</width>
          <height>391</height>
         </rect>
        </property>
        <attribute name="icon">
         <iconset>
          <normaloff>icons/print2.bmp</normaloff>icons/print2.bmp</iconset>
        </attribute>
        <attribute name="label">
         <string>信号发射时机</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="2" column="0" colspan="2">
          <widget class="QTextEdit" name="textEdit"/>
         </item>
         <item row="1" column="1">
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="0">
          <widget class="QToolButton" name="tbtnTextClear">
           <property name="text">
            <string>清空文本</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QToolButton" name="tBtnLineAdd">
           <property name="text">
            <string>添加空行</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>GroupBox</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>当前项变化</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="TextCurrent"/>
        </item>
        <item row="0" column="2">
         <widget class="QCheckBox" name="chkEditable">
          <property name="text">
           <string>可编辑</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QToolButton" name="tBtnSelectItem">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string>tBtnSelectItem</string>
          </property>
          <property name="popupMode">
           <enum>QToolButton::MenuButtonPopup</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="1" colspan="2">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QToolButton" name="tBtnSelectALL">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>tBtnSelectALL</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="tBtnSelectNone">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>tBtnSelectNone</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="tBtnselectInvs">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>tBtnselectInvs</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Preferred</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item row="2" column="0" colspan="3">
         <widget class="QListWidget" name="listWidget">
          <property name="contextMenuPolicy">
           <enum>Qt::CustomContextMenu</enum>
          </property>
          <item>
           <property name="text">
            <string>New Item 1</string>
           </property>
           <property name="checkState">
            <enum>Checked</enum>
           </property>
           <property name="icon">
            <iconset resource="rsc.qrc">
             <normaloff>:/new/prefix1/icons/check2.ico</normaloff>:/new/prefix1/icons/check2.ico</iconset>
           </property>
           <property name="flags">
            <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
           </property>
          </item>
          <item>
           <property name="text">
            <string>New Item 2</string>
           </property>
           <property name="checkState">
            <enum>Checked</enum>
           </property>
           <property name="icon">
            <iconset resource="rsc.qrc">
             <normaloff>:/new/prefix1/icons/check2.ico</normaloff>:/new/prefix1/icons/check2.ico</iconset>
           </property>
           <property name="flags">
            <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actListInit"/>
   <addaction name="actListClear"/>
   <addaction name="actListInsert"/>
   <addaction name="actListAdd"/>
   <addaction name="actListDeleteCurrent"/>
   <addaction name="separator"/>
   <addaction name="actQuit"/>
  </widget>
  <action name="actListInit">
   <property name="icon">
    <iconset resource="rsc.qrc">
     <normaloff>:/new/prefix1/icons/124.bmp</normaloff>:/new/prefix1/icons/124.bmp</iconset>
   </property>
   <property name="text">
    <string>初始化列表</string>
   </property>
   <property name="toolTip">
    <string>初始化列表</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actListClear">
   <property name="icon">
    <iconset resource="rsc.qrc">
     <normaloff>:/new/prefix1/icons/delete1.bmp</normaloff>:/new/prefix1/icons/delete1.bmp</iconset>
   </property>
   <property name="text">
    <string>清除列表</string>
   </property>
   <property name="toolTip">
    <string>清除列表</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actListInsert">
   <property name="icon">
    <iconset resource="rsc.qrc">
     <normaloff>:/new/prefix1/icons/410.bmp</normaloff>:/new/prefix1/icons/410.bmp</iconset>
   </property>
   <property name="text">
    <string>插入项</string>
   </property>
   <property name="toolTip">
    <string>插入项</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actListAdd">
   <property name="icon">
    <iconset resource="rsc.qrc">
     <normaloff>:/new/prefix1/icons/316.bmp</normaloff>:/new/prefix1/icons/316.bmp</iconset>
   </property>
   <property name="text">
    <string>添加项</string>
   </property>
   <property name="toolTip">
    <string>添加项</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actListDeleteCurrent">
   <property name="icon">
    <iconset resource="rsc.qrc">
     <normaloff>:/new/prefix1/icons/324.bmp</normaloff>:/new/prefix1/icons/324.bmp</iconset>
   </property>
   <property name="text">
    <string>删除当前项</string>
   </property>
   <property name="toolTip">
    <string>删除当前项</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actQuit">
   <property name="icon">
    <iconset resource="rsc.qrc">
     <normaloff>:/new/prefix1/icons/exit.bmp</normaloff>:/new/prefix1/icons/exit.bmp</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
   <property name="toolTip">
    <string>退出</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actSelectListALL">
   <property name="text">
    <string>全选</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actSelectListNone">
   <property name="text">
    <string>空选</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actSelectListInverse">
   <property name="text">
    <string>反选</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actListMenu">
   <property name="icon">
    <iconset resource="rsc.qrc">
     <normaloff>:/new/prefix1/icons/128.bmp</normaloff>:/new/prefix1/icons/128.bmp</iconset>
   </property>
   <property name="text">
    <string>选项菜单</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
 </widget>
 <resources>
  <include location="rsc.qrc"/>
 </resources>
 <connections/>
</ui>
